02_snake_animation/main.go

```diff
@@ -48,6 +48,23 @@ func moveSnake(s snake) snake {
 
 // Makes a single iteration for a snake.
 func step(s snake) snake {
+	x, y := s.pos.x, s.pos.y
+	if x == 30 && y == 3 {
+		s.dir.x = 0
+		s.dir.y = 1
+	}
+	if x == 30 && y == 10 {
+		s.dir.x = -1
+		s.dir.y = 0
+	}
+	if x == 3 && y == 10 {
+		s.dir.x = 0
+		s.dir.y = -1
+	}
+	if x == 3 && y == 3 {
+		s.dir.x = 1
+		s.dir.y = 0
+	}
```

Comment:
Nice. To avoid many similar ifs, I'd do the following instead

```golang
for _, pivot := range []struct {
  pos, dir coord
} {
  {coord{30, 3}, coord{0, 1}},
  {coord{30, 10}, coord{-1, 0}},
  {coord{3, 10}, coord{0, -1}},
  {coord{3, 3}, coord{1, 0}},
} {
  if (coord{x, y} == pivot.pos) {
    s.dir = pivot.dir
  }
}
```